### ISSUE TITLE
Copying entire friend ID via custom_packets.

### ISSUE THREAD

**Issue by SkyzohKey** [RicinApp/Ricin#17]
A user should be able to copy friends ToxID. To achieve this I think it would be nice to use the `custom_packets` "method" provided by ToxCore.

**Example:**
- **Alice** right click **Bob** FriendListRow and click on "Ask for ToxID" ;
- **Alice** client send a `friend_request_toxid` packet to **Bob** ;
- **Bob** is asked for sending his entire ToxID (not only the pubkey) ;
  - If **Bob** answer by `Yes` his client send a `friend_response_toxid` containing it's ToxID ;
  - If **Bob** answer by `No` his client send a `friend_response_toxid` containing `null` or `false`.  
- **Alice** client get notified of the `friend_response_toxid` and can:
  - Copy **Bob** ToxID in the Clipboard ;
  - Notify **Alice** that **Bob** rejected her request to get **Bob** entire ToxID.

**Extra:** It should be possible to auto-accept `friend_request_toxid` but this should be disabled by default.
##

**Comment by benwaffle** [comment]
why do you need the other friend's ToxID

**Comment by SkyzohKey** [comment]
It's sometimes useful to have friend ID, for example when you want to add it on another client/computer. Or if a friend demands you the ToxID of another friend ^^

**Comment by ProMcTagonist** [comment]
I believe @subliun is doing something similar for Antox, you should use the same method and be compatible.

**Comment by SkyzohKey** [comment]
ah, nice! @subliun could you talk here about what you did, in order to make Ricin compatible, and maybe adding this in the Tacos/Tox-STS :3

**Comment by Doom032** [comment]
Great idea. Came here from Tox group chat. As a regular Tox user who's been trying to get his friends to use it for a year now(with limited success), I say every Tox client should have this feature. If it can be, it'd be nice if this would be supported in toxcore. One of the most annoying things right now is adding friends of friends. You have to either keep a database of friend IDs like I do, or ask them when they happen to be online.

**Comment by FoundOne** [comment]
I'd like to have such feature. I hate when I have to add every friend's id manually on my phone.

**Comment by benwaffle** [comment]
https://github.com/subliun/Antox/issues/134

**Comment by SkyzohKey** [comment]
@subliun any news about what you did for Antox ?

**Comment by codedust** [comment]
Wouldn't it be nice to have some kind of contact sharing? This could be relatively trivially implemented by displaying a suggestion to make a friend request when someone sends you a Tox ID. Additionally, we could implement a "share contact" button that would automatically ask for another contacts Tox ID (as suggested above) and paste it into the current conversation.

**Comment by SkyzohKey** [comment]
@codedust Yep, should be doable i guess. Can you open an issue for this suggestion please ? x)

**Comment by codedust** [comment]
Done ;)

**Comment by IngvarX** [comment]
You can implement it using plugins like I did in my client (https://github.com/ingvar1995/toxygen_plugins/tree/master/CopyableToxId)
